beetsplug/fetchart.py

```diff
@@ -230,22 +231,42 @@ def fetch_image(self, candidate, extra):
             with closing(self.request(candidate.url, stream=True,
                                       message=u'downloading image')) as resp:
                 ct = resp.headers.get('Content-Type', None)
-                if ct not in CONTENT_TYPES:
-                    self._log.debug(
-                        u'not a supported image: {}',
-                        resp.headers.get('Content-Type') or u'no content type',
-                    )
-                    candidate.path = None
-                    return
-
-                # Generate a temporary file with the correct extension.
-                with NamedTemporaryFile(suffix=b'.' + CONTENT_TYPES[ct][0],
-                                        delete=False) as fh:
-                    for chunk in resp.iter_content(chunk_size=1024):
+
+                # Generate a temporary file and guess the extension based on
+                # the Content-Type header. This may be wrong for badly
+                # configured servers. E.g. fanart.tv only allows .jpg uploads
+                # and ALWAYS returns a image/jpeg Content-Type, but other
+                # formats with a erroneous .jp(e)g extension apparently can
+                # sneak through the upload filter. Therefore validate the type
+                # using the file magic.
+                data = resp.iter_content(chunk_size=1024)
```

Comment:
So I did some quick testing and it seems `chunk_size` is the *maximum* chunk size, and a chunk can be smaller than it.

Here's a quick example I threw together to prove it:

```python
import requests

# Get a 100KiB file from http://speedtest.ftp.otenet.gr/ (i.e. 102400 bytes)
response = requests.get('http://speedtest.ftp.otenet.gr/files/test100k.db')

# Ask for 1MiB from the response (i.e. 1,048,576 bytes)
data = response.iter_content(chunk_size=1048576)

# Get all chunks from the iterator
chunks = list(data)

# Check the number of chunks
print(len(chunks) == 1)

# Check the length of the first chunk
print(len(chunks[0]) == 102400)
```

Both of the print statements output `True`, so we can see that `chunk_size` is the *maximum* size a chunk can be :smile: :tada:.